### PR TITLE
Py2to3

### PR DESCRIPTION
**Pull request by patgmiller**
Added several minor corrections to make django-select2 compatible with Python 2 and 3 using the six library and try catch with ImportError.

### CODE REVIEW

File: django_select2/widgets.py [hunk 1]
```diff
@@ -5,14 +5,18 @@
...
 from django import forms
 from django.core.validators import EMPTY_VALUES
-from django.utils.encoding import force_unicode
 from django.utils.safestring import mark_safe
 from django.core.urlresolvers import reverse
 from django.utils.datastructures import MultiValueDict, MergeDict
+try:
+    from django.utils.encoding import force_unicode
+except ImportError:
+    from django.utils.encoding import force_str as force_unicode
```
Comment (codingjoe) [inline django_select2/widgets.py]: why don't you use force_text?

File: django_select2/util.py [hunk 2]
```diff
@@ -101,7 +106,10 @@ def register_field(key, field):
         if GENERATE_RANDOM_ID:
             id_ = u"%d:%s" % (len(__id_store), unicode(datetime.datetime.now()))
         else:
-            id_ = unicode(hashlib.sha1("%s:%s" % (key, SECRET_SALT)).hexdigest())
+            if six.PY3:
```
Comment (codingjoe) [inline django_select2/util.py]: is this really necessary?

**Comment by codingjoe** [comment]
Hi @patgmiller thanks for the PR, I wrote a couple of comments.
What I can tell you right now, it that it doesn't cover all python3 related problems. What I can't tell you, is it breaks anything in python2.

I'm currently working on a test suit, and hopefully will be done in about 2 weeks. I'll let you know when I'm done, and than we can rebase this PR and see what's missing to support python3.
It that ok with you?

File: django_select2/widgets.py (hunk 1 above)
Comment (patgmiller) [inline django_select2/widgets.py]: I was symploy following what django was using for alias: 

```python
#django.utils.encoding.py

if six.PY3:
    smart_str = smart_text
    force_str = force_text
else:
    smart_str = smart_bytes
    force_str = force_bytes
    # backwards compatibility for Python 2
    smart_unicode = smart_text
    force_unicode = force_text
```

I am fine either way, I'll push a commit to change it.

File: django_select2/util.py (hunk 2 above)
Comment (patgmiller) [inline django_select2/util.py]: Python 3, was throwing an error that strings needed to be encoded before hashing them. I figured it was simpler to leave Python 2 implementation as is, and just fix the Python 3 way. Would you prefer and single uniform implementation?

**Comment by patgmiller** [comment]
Waiting until the test suite is ready and rebasing is fine with me. I wouldn't want to break anything. I can function off my fork as a temporary solution and revisit it when the test suite is ready. Thanks,

File: django_select2/util.py [hunk 3]
```diff
@@ -101,7 +106,7 @@ def register_field(key, field):
         if GENERATE_RANDOM_ID:
             id_ = u"%d:%s" % (len(__id_store), unicode(datetime.datetime.now()))
         else:
-            id_ = unicode(hashlib.sha1("%s:%s" % (key, SECRET_SALT)).hexdigest())
+            id_ = hashlib.sha1(six.text_type("%s:%s" % (key, SECRET_SALT)).encode()).hexdigest()
```
Comment (patgmiller) [inline django_select2/util.py]: @codingjoe You are correct it is not necessary to have two different lines for this. I did a quick check in Python 2.7 and Python 3.4. 

```python
patrick@serenity:django-select2 (local)$ ipython
Python 2.7.6 (default, Mar 22 2014, 22:59:56) 
Type "copyright", "credits" or "license" for more information.

IPython 1.2.1 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: import hashlib, six

In [2]: s =six.text_type("%s:%s" % ('key', 'secret salt')).encode()

In [3]: hashlib.sha1(s).hexdigest()
Out[3]: 'a4fb0f227e2b7fcfdf5c50a6ea42d9998144eec3'
```

```python
Python 3.4.0 (default, Apr 11 2014, 13:05:11) 
Type "copyright", "credits" or "license" for more information.

IPython 2.3.1 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: import hashlib, six

In [2]: s =six.text_type("%s:%s" % ('key', 'secret salt')).encode()

In [3]: hashlib.sha1(s).hexdigest()
Out[3]: 'a4fb0f227e2b7fcfdf5c50a6ea42d9998144eec3'
```

Comment (kevin-brown) [inline django_select2/util.py]: Can't you just use `force_unicode`/`force_text` here instead of calling `six.text_type`?

**Comment by codingjoe** [comment]
I just refactored the entire code base to work in both python 2 and python 3.
Thanks for your contrib.

**Pull request closed without merging**